src/sources/datadog_agent/logs.rs

```diff
@@ -192,3 +201,71 @@ pub(crate) fn decode_log_body(
 
     Ok(decoded)
 }
+
+// ddtags input is a string containing a list of key-value pairs.
+// the tag-value list members are separated by `,` and the
+// tag-value pairs are separated by `:`.
+//
+// The output is an Object constructed with the tag-value pairs.
+fn parse_ddtags(ddtags_raw: &Bytes) -> ObjectMap {
+    let ddtags_str = String::from_utf8_lossy(ddtags_raw);
+
+    let mut ddtags_object = ObjectMap::new();
+
+    ddtags_str.split(',').for_each(|kv| {
+        if let Some((k, v)) = kv.split_once(':') {
+            ddtags_object.insert(KeyString::from(k), Value::Bytes(Bytes::from(v.to_string())));
+        }
+    });
+
+    if ddtags_object.is_empty() && !ddtags_str.is_empty() {
+        warn!(message = "`parse_ddtags` set to true and Agent log contains non-empty ddtags string, but no tag-value pairs were parsed.")
```

Comment:
I don't know that the bit about `parse_ddtags` being set adds much value to this warning.
```suggestion
        warn!(message = "Agent log contains non-empty `ddtags` string, but no tag-value pairs were parsed.");
```

Reply:
FWIW I like the mention of the flag

Reply:
🤷  I'm ambivalent. I left it in but if you feel strongly Bruce I can remove it.

Reply:
Don't care that strongly.